Scripts/WikiUpdater.cs

```diff
@@ -0,0 +1,415 @@
+                        .Select(li => $"[indent]—   {ConvertTextToBbcode(li.InnerHtml)}[/indent]")
+                        .Aggregate((a, b) => a + "\n" + b) + "\n\n";
+                    break;
+                default:
+                    // Ignore all other tag types
+                    continue;
+            }
+
+            // Concatenate this tag with the rest of the section so far
+            sections[^1] = new Wiki.Page.Section(sections[^1].SectionHeading, sections[^1].SectionBody + text);
+        }
+
+        return sections.Select(s => new Wiki.Page.Section(s.SectionHeading, s.SectionBody.Trim())).ToList();
+    }
+
+    /// <summary>
+    ///   Returns an equivalent section of a wiki page where the heading and body have been replaced with appropriate
+    ///   translation keys.
+    /// </summary>
+    private static Wiki.Page.Section UntranslateSection(Wiki.Page.Section section, string pageName)
+    {
+        var sectionName = section.SectionHeading?.ToUpperInvariant().Replace(" ", "_");
+        var heading = sectionName != null ? $"WIKI_HEADING_{sectionName}" : null;
+        var body = sectionName != null ? $"WIKI_{pageName}_{sectionName}" : $"WIKI_{pageName}_INTRO";
+
+        return new Wiki.Page.Section(heading, body);
+    }
+
+    /// <summary>
+    ///   Converts HTML for a single paragraph into BBCode. Paragraph must not contain lists, headings, etc.
+    /// </summary>
+    private static string ConvertTextToBbcode(string paragraph)
+    {
+        // Process our custom BBCode first
+        foreach (var compound in CustomBbcodeCompounds)
+        {
+            var compoundText = compound.ToLowerInvariant().Replace(" ", string.Empty);
+            paragraph = paragraph.Replace(
+                $"<b>{compound}</b>",
+                $"[thrive:compound type=\"{compoundText}\"][/thrive:compound]");
+        }
+
+        return paragraph
+            .Replace("\n", string.Empty)
+            .Replace("<b>", "[b]")
+            .Replace("</b>", "[/b]")
+            .Replace("<i>", "[i]")
+            .Replace("</i>", "[/i]")
+            .Replace("\"", "\\\"");
+    }
+
+    /// <summary>
+    ///   Inserts into en.po the English translations for all the translation keys in a list of wiki pages.
+    /// </summary>
+    private static async Task InsertTranslatedPageContent(List<TranslationPair> pages,
+        CancellationToken cancellationToken)
+    {
+        // Create the whole list of values to replace first, then replace asynchronously based on read lines
+        var translationPairs = new Dictionary<string, string>();
+
+        foreach (var page in pages)
+        {
+            var untranslatedPage = page.UntranslatedPage;
+            var translatedPage = page.TranslatedPage;
+
+            // Translate page names
+            translationPairs.Add(untranslatedPage.Name, translatedPage.Name);
+
+            for (var j = 0; j < untranslatedPage.Sections.Count; j++)
```

Comment:
Is there a reason why this uses `j`? I can't see a higher level loop that already used `i` so the first loop using an index variable should use `i`.

Reply:
Sorry, that was an artifact of the nested loops in the previous algorithm that I'd modified. Fixed in both places now.